Vostok.ClusterConfig.Client/Helpers/DefaultClusterConfigClientProvider.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using JetBrains.Annotations;
+
+namespace Vostok.ClusterConfig.Client.Helpers
+{
+    internal static class DefaultClusterConfigClientProvider
+    {
+        private static readonly object Sync = new object();
+
+        private static volatile ClusterConfigClient client;
+
+        public static ClusterConfigClient Get()
+        {
+            if (client != null)
+                // ReSharper disable once InconsistentlySynchronizedField
+                return client;
+
+            lock (Sync)
+            {
+                client = client ?? Create();
+                return client;
+            }
+        }
+
+        public static void Configure([NotNull] ClusterConfigClient newClient, bool canOverwrite = false)
+        {
+            if (newClient == null)
+                throw new ArgumentNullException(nameof(newClient));
+
+            lock (Sync)
+            {
+                if (!canOverwrite && client != null)
+                    throw new InvalidOperationException("Can't overwrite existing ClusterConfigClient.");
+
+                client = newClient;

```

Comment:
What are the scenarios where it would be benificial to overwrite an already initialized client?

How do we handle its lifetime then? Should we dispose of the instance being swapped out?

Reply:
> Should we dispose of the instance being swapped out?
We can't dispose old instance, because someone can already setup configuration provider with it.
But maybe sometimes we want to overwrite it. 
Also it can be manually disposed via `.Default.Dispose` and reconfigured again.